web/src/main/java/org/wso2/testgrid/web/api/LogEventSocketMediator.java

```diff
@@ -0,0 +1,197 @@
+            throws TestGridException, IOException, FileWatcherException {
+        Path logFilePath = getLogFilePath(testPlan);
+        FileWatcher fileWatcher = new LogFileWatcher(logFilePath, session);
+        new Thread(fileWatcher).start();
+    }
+
+    /**
+     * Returns the path of the log file for the given test plan.
+     *
+     * @param testPlan test plan to read log file from
+     * @return path of the log file related to the test plan
+     * @throws TestGridException thrown when error on calculating the log file path
+     */
+    private Path getLogFilePath(TestPlan testPlan) throws TestGridException, IOException {
+        return Paths.get(TestGridUtil.getTestGridHomePath(), TestGridUtil.getTestRunArtifactsDirectory(testPlan)
+                .resolve(TestGridConstants.TEST_LOG_FILE_NAME).toString());
+    }
+
+    /**
+     * This class is responsible for watching changes of the given log file.
+     *
+     * @since 1.0.0
+     */
+    private static class LogFileWatcher extends FileWatcher {
+
+        private final Session session;
+        private String currentFileContents = "";
+
+        /**
+         * Creates an instance of {@link LogFileWatcher} to watch the given file.
+         *
+         * @param watchFile file to be watched
+         * @param session   client session
+         * @throws FileWatcherException thrown when error on creating an instance of {@link LogFileWatcher}
+         */
+        LogFileWatcher(Path watchFile, Session session) throws FileWatcherException {
+            super(watchFile);
+            this.session = session;
+        }
+
+        @Override
+        public void beforeFileWatch(String fileContents) throws FileWatcherException {
+            sendTextDiffToClient(fileContents);
+        }
+
+        @Override
+        public void onCreate(String fileContents) throws FileWatcherException {
+            sendTextDiffToClient(fileContents);
+        }
+
+        @Override
+        public void onModified(String fileContents) throws FileWatcherException {
+            sendTextDiffToClient(fileContents);
+        }
+
+        /**
+         * Sends the file contents diff to client.
+         *
+         * @param fileContents file contents
+         * @throws FileWatcherException thrown when error on sending file contents to the client
+         */
+        private void sendTextDiffToClient(String fileContents) throws FileWatcherException {
+            try {
+                String difference = org.apache.commons.lang3.StringUtils.difference(currentFileContents, fileContents);
+                currentFileContents = fileContents;
+                session.getBasicRemote().sendText(difference);
+            } catch (IOException e) {
+                throw new FileWatcherException("Error on sending file contents to the client.", e);
+            }
+        }
+
+        @Override
+        public void onDelete() {
+            // Do nothing
```

Comment:
do we need onDelete method of inherited class itself?

Reply:
This is for the completeness of the `FileWatcher` class. We do not need this at the moment. However, in future, we may need to identify these events as well.